ICSharpCode.Decompiler/TypeSystem/MetadataModule.cs

```diff
@@ -491,75 +491,70 @@ IMethod ResolveMethodReference(MemberReferenceHandle memberRefHandle, GenericCon
 				var statedDeclaringType = ResolveDeclaringType(memberRef.Parent, context);
 				var statedDeclaringTypeDefinition = statedDeclaringType.GetDefinition();
 
-				// Note: statedDeclaringType might be parameterized, but the signature is for the original method definition.
-				// We'll have to search the member directly on statedDeclaringTypeDefinition.
-				signature = memberRef.DecodeMethodSignature(TypeProvider,
-					new GenericContext(statedDeclaringTypeDefinition?.TypeParameters));
-
-				if (statedDeclaringTypeDefinition != null)
+				// GetAllBaseTypes() returns types ordered from base -> specific. Traverse the list in reverse
+				// to check in the opposite direction (most specific to least specific).
+				var candidateDeclaringTypes = statedDeclaringType.GetAllBaseTypes().ToList();
```

Comment:
Is it correct to search all base types? I would assume the runtime only searches base classes, not interfaces.